### PR TITLE
Added LogHandler, this enables the possibility to change the behavior when logging messages with this library.

### PR DESCRIPTION
**Pull request by urmahp**
I have renamed the log macros to something more specific to this library, this make the macros less likely to clash with other macros in the project, where this library is used. 

I have added a default log handler which is used as default and as long as no other loghandler is enabled.

### CODE REVIEW

**Comment by fmauch** [comment]
Quickly skimming over the changes this looks good to me. It might be a good idea to factor out the ROS logger, as well and move it to the driver, but we can do that in a separate build, as well.

@urmahp I guess, this is an alternative to #32?

**Comment by urrsk** [comment]
@fmauch yes it is an alternative to #32.
1. The plan is to first implement this in the ISAAC driver
2. Then we can see when to merge it into the ros driver. (Maybe introduce it in the beta or wait to after release)

@fmauch To not take too much of your time, I will ask @t-schnell to review it. But you are more than welcome.

**Comment by fmauch** [comment]
I've tested it yesterday with a minimal example I have, where I created my own logger. It worked well.

I think, it would be good to also add some documentation on how to include this into your own application.

Also, it might be good to set the default logging mechanism to WARN to reduce the default amount of output.

**Comment by urmahp** [comment]
Thanks for the contribution @fmauch, that is a good idea. 

I will update the pr today.

**Review by fmauch — CHANGES_REQUESTED** [review]

File: include/ur_client_library/log.h
Comment (fmauch) [reply include/ur_client_library/log.h:81]: I've got it implemented locally, I can make a PR to your branch later today...

Comment (urmahp) [reply include/ur_client_library/log.h:81]: That would be nice, thank you very much. 

I should have checked, so that maintainers should be able to commit directly to this branch, so I guess you don't have to make a PR.

Comment (fmauch) [reply include/ur_client_library/log.h:81]: I created a PR nevertheless so you can check it.

**Review by fmauch — APPROVED** [review]
Looks good to me. I'm holding this back, as the ROS driver needs a patch before merging this. A couple of `LOG_XXXX` macros have made it into the driver which will result in build failures.

**Comment by fmauch** [comment]
I've rebased and cleaned up this branch. As soon as CI runs through, I'll merge in those changes.

**Comment by fmauch** [comment]
Thanks @urmahp for this contribution!

**Pull request merged**
